src/details/ArborX_IndexableGetter.hpp

```diff
@@ -40,19 +40,22 @@ struct DefaultIndexableGetter
   }
 };
 
-template <typename Primitives>
+template <typename Values, typename IndexableGetter>
 struct Indexables
 {
-  Primitives _primitives;
-  using Access = AccessTraits<Primitives, PrimitivesTag>;
+  Values _values;
+  IndexableGetter _indexable_getter;
+
+  using Access = AccessTraits<Values, PrimitivesTag>;
   using memory_space = typename Access::memory_space;
 
-  KOKKOS_FUNCTION decltype(auto) operator()(int i) const
+  // FIXME: why is using `delctype(auto)` or `auto const &` here not work?
+  KOKKOS_FUNCTION auto operator()(int i) const
```

Comment:
@dalg24 I need your help to figure this out. I don't understand why `decltype(auto)` does not work here, returning completely wrong stuff.

Reply:
That should work. I checked on my local build and on Crusher and it did.

Reply:
> That should work. I checked on my local build and on Crusher and it did.

What should work? `decltype(auto)` does not work on Crusher. For example, if you run DBSCAN benchmark with it, it takes couple orders of magnitude longer due to the wrong hierarchy.